main/guides/getting-started/contract-rpc.md

```diff
@@ -0,0 +1,410 @@
+    },
+  ],
+}
+```
+
+Then we can use `agoric follow` without any options to
+get a stream of updates as they appear.
+
+```sh
+agoric follow :published.wallet.agoric1h4d3mdvyqhy2vnw2shq4pm5duz5u8wa33jy6cl
+...
+{
+  status: {
+    id: "closeVault-1691526597848",
+    invitationSpec: {
+      invitationMakerName: "CloseVault",
+      previousOffer: "openVault-1691526589332",
+      source: "continuing",
+    },
+    numWantsSatisfied: 1,
+    payouts: {
+      Collateral: {
+        brand: slotToVal("board05557","Alleged: ATOM brand"),
+        value: 13000000n,
+      },
+      Minted: {
+        brand: slotToVal("board0257","Alleged: IST brand"),
+        value: 215000n,
+      },
+    },
+    proposal: {
+      give: {
+        Minted: {
+          brand: slotToVal("board0257"),
+          value: 5750000n,
+        },
+      },
+      want: {},
+    },
+    result: "your vault is closed, thank you for your business",
+  },
+  updated: "offerStatus",
+}
+```
+
+Note that status updates are emitted at several points in the handling of
+each offer:
+
+- when the `getOfferResult()` promise settles
+- when the `numWantsSatisfied()` promise settles
+- when the payouts have been deposited.
+
+And we may get `balance` updates at any time.
+
+The data published via vstorage are available within the JavaScript VM
+via the [getPublicTopics](https://github.com/Agoric/agoric-sdk/blob/mainnet1B/packages/smart-wallet/src/smartWallet.js#L585) API.
+
+The [CurrentWalletRecord](https://github.com/Agoric/agoric-sdk/blob/mainnet1B/packages/smart-wallet/src/smartWallet.js#L71-L76) type is:
+
+```ts
+{
+   purses: Array<{brand: Brand, balance: Amount}>,
+   offerToUsedInvitation: Array<[ offerId: string, usedInvitation: Amount ]>,
+   offerToPublicSubscriberPaths: Array<[ offerId: string, publicTopics: { [subscriberName: string]: string } ]>,
+   liveOffers: Array<[import('./offers.js').OfferId, import('./offers.js').OfferStatus]>,
+}
+```
+
+And [UpdateRecord](https://github.com/Agoric/agoric-sdk/blob/mainnet1B/packages/smart-wallet/src/smartWallet.js#L80-L83) is:
+
+```ts
+     { updated: 'offerStatus', status: import('./offers.js').OfferStatus }
+   | { updated: 'balance'; currentAmount: Amount }
+   | { updated: 'walletAction'; status: { error: string } }
+```
+
+Both of those types include [OfferStatus](https://github.com/Agoric/agoric-sdk/blob/mainnet1B/packages/smart-wallet/src/offers.js#L21C14-L26C5) by reference:
+
+```ts
+import('./offers.js').OfferSpec & {
+ error?: string,
+ numWantsSatisfied?: number
+ result?: unknown | typeof UNPUBLISHED_RESULT,
+ payouts?: AmountKeywordRecord,
+}
+```
+
+## VBank Assets and Cosmos Bank Balances
+
+Note that balances of assets such as **IST** and **BLD** are already
+available via consensus layer queries to the Cosmos SDK [bank module](https://docs.cosmos.network/main/modules/bank).
+
+```sh
+$ agd query bank balances agoric1h4d3mdvyqhy2vnw2shq4pm5duz5u8wa33jy6cl -o json | jq .balances
```

Comment:
@Wietze wrote:

> What might be a help for new developers is to provide a clear example for querying the balances of vbank assets (IST etc), now it feels a bit disjointed to have to go to the CosmosSDK to do the rest (most Agoric implementations fully abstract it away which is I think a good thing). Potentially pointing them to where this happens in wallet-UI might be enough. 

> Will there be some kind of wrapper within the agoric-sdk for this in the future potentially 

I replied:

I could see us factoring out such a wrapper... Somewhere in the ui-kit repo, perhaps.

Querying balances with just fetch is pretty straightforward... I suppose an example could be pretty small. But the page is getting long... I might want to split it up. Hm.